test/services/envoy/factory.go

```diff
@@ -0,0 +1,176 @@
+package envoy
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/solo-io/skv2/codegen/util"
+
+	"github.com/solo-io/gloo/test/testutils"
+	"github.com/solo-io/gloo/test/testutils/version"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	errors "github.com/rotisserie/eris"
+
+	"github.com/solo-io/go-utils/log"
+)
+
+type Factory interface {
+	MustEnvoyInstance() *Instance
+	NewEnvoyInstance() (*Instance, error)
+	MustClean()
+}
+
+var _ Factory = new(factoryImpl)
+
+type factoryImpl struct {
+	instanceManager *InstanceManager
```

Comment:
This seems like a good candidate to use composition. 
```suggestion
	*InstanceManager
```